conda/common/logic.py

```diff
@@ -778,6 +774,14 @@ def itersolve(self, constraints=None, m=None):
             exclude.append([-k for k in sol if -m <= k <= m])
 
     def minimize(self, objective, bestsol=None, trymax=False):
+        if isinstance(objective, dict):
+            objective_list = [(v, self.names.get(k, k)) for k, v in objective.items()]
+        else:
+            objective_list = objective
+
+        return self._minimize(objective_list, bestsol=bestsol, trymax=trymax)
+
+    def _minimize(self, objective, bestsol=None, trymax=False):
```

Comment:
The idea here is to separate all "variable name handling" from the rest of the logic.
https://github.com/conda/conda/pull/9427/commits/f6d191f3f1ee9f7348f6b3f5a079646707b8099e has the bulk of those changes.